packages/data/src/lookitS3.ts

```diff
@@ -27,13 +28,22 @@ class LookitS3 {
    */
   public constructor(key: string) {
     this.key = key;
-    this.s3 = new S3Client({
-      region: process.env.S3_REGION,
-      credentials: {
-        accessKeyId: process.env.S3_ACCESS_KEY_ID,
-        secretAccessKey: process.env.S3_SECRET_ACCESS_KEY,
-      },
-    });
+    try {
+      this.s3 = new S3Client({
+        region: process.env.S3_REGION,
+        credentials: {
+          accessKeyId: process.env.S3_ACCESS_KEY_ID,
+          secretAccessKey: process.env.S3_SECRET_ACCESS_KEY,
+        },
+      });
+    } catch (e) {
+      console.error(`Error setting up S3 client: ${e}`);
+      let err_msg = "";
+      if (e instanceof Error) {
+        err_msg = e.message;
+      }
+      throw new AWSConfigError(err_msg);
```

Comment:
This is not necessarily wrong; I don't understand why this is needed.  Could you tell me more about why we must check if `message` is on the error object? Could we move this logic into the error class?

Reply:
Yeah, this is ugly. I tried to just catch the error but also pass it to `AWSConfigError` so that we'd have more detailed info on what the problem was, like this:
```typescript
} catch (e) {
  console.error(`Error setting up S3 client: ${e.message}`);
  throw new AWSConfigError(e.message);
}
```
But TypeScript doesn't allow this because the argument passed to `catch` is of type unknown (not Error), so I can't get the message property without checking the type (see https://stackoverflow.com/a/54649623).
But yes, I can definitely pass this into the Error class as unknown and move the logic, if that would be better?

Reply:
I see what you're saying now.  Let's leave this for another day.